internal/backend/azure/azure.go

```diff
@@ -101,6 +101,18 @@ func open(cfg Config, rt http.RoundTripper) (*Backend, error) {
 		if err != nil {
 			return nil, errors.Wrap(err, "NewAccountSASClientFromEndpointToken")
 		}
+	} else if cfg.ForceCliCredential {
+		debug.Log(" - using AzureCLICredential")
+
+		cred, err := azidentity.NewAzureCLICredential(nil)
```

Comment:
The only difference to the `- using DefaultAzureCredential` case is how the credentials are obtained. Please merge both cases.

Reply:
I don't have much experience in Go yet, I was trying this but it fails with "cred declared and not used". Any idea?

```go
	} else {
		var cred *azcore.TokenCredential
		if cfg.ForceCliCredential {
			debug.Log(" - using AzureCLICredential")
			cred, err := azidentity.NewAzureCLICredential(nil)
			if err != nil {
				return nil, errors.Wrap(err, "NewAzureCLICredential")
			}
		} else {
			debug.Log(" - using DefaultAzureCredential")
			cred, err := azidentity.NewDefaultAzureCredential(nil)
			if err != nil {
				return nil, errors.Wrap(err, "NewDefaultAzureCredential")
			}
		}

		client, err = azContainer.NewClient(url, *cred, opts)
		if err != nil {
			return nil, errors.Wrap(err, "NewClient")
		}
	}
```

Reply:
I've made the necessary change.

 `:=` declares a new variable within the current scope. That is `cred` in `var cred ...` and `cred, err := ...` are different variables.